extensions-core/druid-catalog/src/test/java/org/apache/druid/server/http/catalog/CatalogResourceTest.java

```diff
@@ -0,0 +1,697 @@
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.druid.server.http.catalog;
+
+import org.apache.druid.catalog.http.CatalogResource;
+import org.apache.druid.catalog.http.MoveColumn;
+import org.apache.druid.catalog.http.TableEditRequest;
+import org.apache.druid.catalog.http.TableEditRequest.DropColumns;
+import org.apache.druid.catalog.http.TableEditRequest.HideColumns;
+import org.apache.druid.catalog.http.TableEditRequest.UnhideColumns;
+import org.apache.druid.catalog.model.CatalogUtils;
+import org.apache.druid.catalog.model.Columns;
+import org.apache.druid.catalog.model.TableId;
+import org.apache.druid.catalog.model.TableMetadata;
+import org.apache.druid.catalog.model.TableSpec;
+import org.apache.druid.catalog.model.table.AbstractDatasourceDefn;
+import org.apache.druid.catalog.model.table.DatasourceDefn;
+import org.apache.druid.catalog.model.table.InlineTableDefn;
+import org.apache.druid.catalog.model.table.InputFormats;
+import org.apache.druid.catalog.model.table.TableBuilder;
+import org.apache.druid.catalog.storage.CatalogTests;
+import org.apache.druid.metadata.TestDerbyConnector;
+import org.apache.druid.server.security.ForbiddenException;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+
+import javax.ws.rs.core.Response;
+
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+import static org.apache.druid.server.http.catalog.DummyRequest.deleteBy;
+import static org.apache.druid.server.http.catalog.DummyRequest.getBy;
+import static org.apache.druid.server.http.catalog.DummyRequest.postBy;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNull;
+import static org.junit.Assert.assertThrows;
+import static org.junit.Assert.assertTrue;
+
+/**
+ * Test of REST API operations for the table catalog.
+ */
+public class CatalogResourceTest
+{
+  @Rule
+  public final TestDerbyConnector.DerbyConnectorRule derbyConnectorRule = new TestDerbyConnector.DerbyConnectorRule();
+
+  private CatalogTests.DbFixture dbFixture;
+  private CatalogResource resource;
+
+  @Before
+  public void setUp()
+  {
+    dbFixture = new CatalogTests.DbFixture(derbyConnectorRule);
+    resource = new CatalogResource(dbFixture.storage, CatalogTests.AUTH_MAPPER);
+  }
+
+  @After
+  public void tearDown()
+  {
+    CatalogTests.tearDown(dbFixture);
+  }
+
+  private static long getVersion(Response resp)
+  {
+    @SuppressWarnings("unchecked")
+    Map<String, Object> result = (Map<String, Object>) resp.getEntity();
+    return (Long) result.get("version");
+  }
+
+  @Test
+  public void testCreate()
```

Comment:
would be nice to split, so that each test is one function, easier to follow the tests, and what is tested. Also with this, the names of the test functions can clearly communicate what is being tested: for example:


```
@Test
  public void given_emptySchemaName_when_postTable_then_throwBadRequest()
  {
    final String tableName = "create";
    TableSpec dsSpec = TableBuilder.datasource(tableName, "P1D").buildSpec();

    // Blank schema name: infer the schema.
    Response resp = resource.postTable("", tableName, dsSpec, 0, false, postBy(CatalogTests.SUPER_USER));
    assertEquals(Response.Status.BAD_REQUEST.getStatusCode(), resp.getStatus());
}
```

If you agree, can do in follow up.